src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -385,6 +384,11 @@ internal void SerializeEncoding(SerializationInfo info, StreamingContext context
             }
         }
 
+        // On Desktop, encoding instances that aren't cached in a static field are cached in
+        // a hash table by codepage.
+        private static volatile Hashtable encodings;
```

Comment:
I agree that it is best to not churn the desktop implementation as part of this PR.